library/init/data/nat/bitwise.lean

```diff
@@ -11,15 +11,78 @@ universe u
 
 namespace nat
 
+  def bodd_div2 : ℕ → bool × ℕ
+  | 0        := (ff, 0)
+  | (succ n) :=
+    match bodd_div2 n with
+    | (ff, m) := (tt, m)
+    | (tt, m) := (ff, succ m)
+    end
+
+  def div2 (n : ℕ) : ℕ := (bodd_div2 n).2
+
+  def bodd (n : ℕ) : bool := (bodd_div2 n).1
```

Comment:
> I didn't think it would be important, but it kept breaking proofs in weird ways.

Do you still remember what it broke?  As far as I can tell, `bodd_div2 n` only reduces if `n` is a numeral.  But in this case I would have expected the WF definition to reduce as well.